deeppavlov/models/multitask_pal_bert/modeling.py

```diff
@@ -0,0 +1,700 @@
+        hidden_states = self.dropout(hidden_states)
+        hidden_states = self.LayerNorm(hidden_states + input_tensor)
+        return hidden_states
+
+
+class BERTSelfOutput(nn.Module):
+    def __init__(self, config, multi_params=None, houlsby=False):
+        super(BERTSelfOutput, self).__init__()
+        if houlsby:
+            multi = BERTLowRank(config)
+            self.multi_layers = nn.ModuleList(
+                [copy.deepcopy(multi) for _ in range(config.num_tasks)]
+            )
+        if multi_params is not None:
+            self.dense = nn.Linear(
+                config.hidden_size_aug, config.hidden_size_aug)
+        else:
+            self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.LayerNorm = BERTLayerNorm(config, multi_params)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+        self.houlsby = houlsby
+
+    def forward(self, hidden_states, input_tensor, attention_mask=None, i=0):
+        hidden_states = self.dense(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        if self.houlsby:
+            hidden_states = hidden_states + self.multi_layers[i](
+                hidden_states, attention_mask
+            )
+        hidden_states = self.LayerNorm(hidden_states + input_tensor)
+        return hidden_states
+
+
+class BERTAttention(nn.Module):
+    def __init__(self, config, multi_params=None, houlsby=False):
+        super(BERTAttention, self).__init__()
+        self.self = BERTSelfAttention(config, multi_params)
+        self.output = BERTSelfOutput(config, multi_params, houlsby)
+
+    def forward(self, input_tensor, attention_mask, i=0):
+        self_output = self.self(input_tensor, attention_mask)
+        attention_output = self.output(
+            self_output, input_tensor, attention_mask, i=i)
+        return attention_output
+
+
+class BERTPals(nn.Module):
+    def __init__(self, config, extra_dim=None):
+        super(BERTPals, self).__init__()
+        # Encoder and decoder matrices project down to the smaller dimension
+        self.aug_dense = nn.Linear(config.hidden_size, config.hidden_size_aug)
+        self.aug_dense2 = nn.Linear(config.hidden_size_aug, config.hidden_size)
+        # Attention without the final matrix multiply.
+        self.attn = BERTSelfAttention(config, 6)
+        self.config = config
+        self.hidden_act_fn = gelu
+
+    def forward(self, hidden_states, attention_mask=None):
+        hidden_states_aug = self.aug_dense(hidden_states)
+        hidden_states_aug = self.attn(hidden_states_aug, attention_mask)
+        hidden_states = self.aug_dense2(hidden_states_aug)
+        hidden_states = self.hidden_act_fn(hidden_states)
+        return hidden_states
+
+
+class BERTLowRank(nn.Module):
+    def __init__(self, config, extra_dim=None):
+        super(BERTLowRank, self).__init__()
+        # Encoder and decoder matrices project down to the smaller dimension
+        if config.extra_dim:
```

Comment:
You can define by if/else only variable hidden_size_aug and define these linear layers out of if/else

Reply:
> You can define by if/else only variable hidden_size_aug and define these linear layers out of if/else

Okay